ledger/src/blockstore.rs

```diff
@@ -3809,9 +3809,16 @@ macro_rules! get_tmp_ledger_path_auto_delete {
 }
 
 pub fn get_ledger_path_from_name_auto_delete(name: &str) -> TempDir {
-    let path = get_ledger_path_from_name(name);
+    let mut path = get_ledger_path_from_name(name);
+    // path is a directory so .file_name() returns the last component of the path
+    let last = path.file_name().unwrap().to_str().unwrap().to_string();
+    path.pop();
     fs::create_dir_all(&path).unwrap();
-    TempDir::new_in(path).unwrap()
+    Builder::new()
+        .prefix(&last)
```

Comment:
Each test makes a new folder that has filename & line# in it as identifying information. We would then make a temporary directory inside of there; this would get cleaned up but we'd still have the outer directory.

Popping the last element off the path and using it as prefix instead makes the directory with filename & line# the TempDir itself, which means this extra directory also gets cleaned up.